ddtrace/span.py

```diff
@@ -163,12 +177,15 @@ def duration(self):
         """The span duration in seconds."""
         if self.duration_ns is not None:
             return self.duration_ns / 1e9
```

Comment:
I think this is also problematic type-wise. The setter assumes the value being set is an `int`, but it comes back as a `float`. I'd do
~~~ python
return self.duration_ns // int(1e9)
~~~

Reply:
If I'm not mistaken, `self.duration` is the duration in seconds (can be a float) while `self.duration_ns` is the integer value in nanoseconds. Since this is just returning the duration (which can be float), this should be fine 😄

Reply:
Ah ok that makes sense; then the type hint for the setter needs to be fixed I suppose?

Reply:
> Since this is just returning the duration (which can be float), this should be fine 😄
If that's correct though, this means that the setter should accept `float` arguments for the `value` being set. I'll make the corresponding changes 😄